backend/webserver/tests.py

```diff
@@ -620,3 +623,621 @@ def test_delete_404(self):
+
+    def test_get_404(self):
+        author_1 = Author.objects.create(username="author_1", display_name="author_1")
+        fake_post_id = 590385093845945
+        url = f'/api/authors/{author_1.id}/posts/{fake_post_id}/'
+        self.client.force_authenticate(user=mock.Mock())
+        response = self.client.get(url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+    
+
+
+    def test_edit_posts(self):
+        """POST request works on all editable data fields"""
+        author_1 = Author.objects.create() 
+        author_1.set_password("pass123")
+        author_1.save()
+        request_payload = {"username": "author_1", "password": "pass123"}
+        response = self.client.post("/login/", data=request_payload, format="json")
+        current_date_string = datetime.datetime.utcnow().replace(tzinfo=utc)
+        post_1 = Post.objects.create(
+            author =author_1,
+            created_at=current_date_string,
+            edited_at=current_date_string,
+            title="Test Post",
+            description="Testing post",
+            source="source",
+            origin="origin",
+            unlisted=False,
+            content_type= "text/plain",
+            content="Some content",
+            visibility= "PUBLIC"
+        )
+        url = f'/api/authors/{author_1.id}/posts/{post_1.id}/'
+        self.client.force_authenticate(user=author_1)
+        payload = {
+            "title": "Mark McGoey",
+            "description": "new description",
+            "unlisted":True,
+            "content":"Some new content"
+        }
+        response = self.client.post(url,data=payload)
+        self.assertEqual(response.data["title"], "Mark McGoey")
+        self.assertEqual(response.data["description"], "new description")
+        self.assertEqual(response.data["unlisted"], True)
+        self.assertEqual(response.data["content"], "Some new content")
+
+
+    def test_non_editable_fields(self):
+        author_1 = Author.objects.create() 
+        current_date_string = datetime.datetime.utcnow().replace(tzinfo=utc)
+        post_1 = Post.objects.create(
+            author =author_1,
+            created_at=current_date_string,
+            edited_at=current_date_string,
+            title="Test Post",
+            description="Testing post",
+            source="source",
+            origin="origin",
+            unlisted=False,
+            content_type= "text/plain",
+            content="Some content",
+            visibility= "PUBLIC"
+        )
+        url = f'/api/authors/{author_1.id}/posts/{post_1.id}/'
+        self.client.force_authenticate(user=mock.Mock())
+        payload = {
+            "source":"new source",
+            "origin": "new origin",  
+            "visibility":"FRIENDS"  
+        }
+        self.client.post(url,data=payload)
+        response = self.client.get(url,format=json)
```

Comment:
What is the response status code in this case? Ideally, the status should be bad request otherwise the client has no way of knowing whether the update went through from the response. They will just assume that the update went through if you return status 200 OK.